pkg/internal/codegen/parse/crd.go

```diff
@@ -188,9 +188,12 @@ func (b *APIs) typeToJSONSchemaProps(t *types.Type, found sets.String, comments
 	case quantity:
 		specialTypeProps.Type = "string"
 		return specialTypeProps, b.getQuantity()
-	case meta, unstructured, rawExtension:
+	case unstructured, rawExtension:
 		specialTypeProps.Type = "object"
 		return specialTypeProps, b.objSchema()
+	case meta:
+		// Validating ObjectMeta as part of https://github.com/kubernetes-sigs/controller-tools/issues/167
+		return b.parseNativeObjectValidation(t, found, comments, isRoot)
```

Comment:
If you pass `specialTypeProps` here, could you remove the the re-execution of `parseDescription` and `getValidation` in `parseNativeObjectValidation` (line 480-487)?